src/main/java/org/fcrepo/migration/handlers/BasicObjectVersionHandler.java

```diff
@@ -174,13 +162,273 @@ public void processObjectVersions(Iterable<ObjectVersionReference> versions) {
+     *
+     * @param triplesToRemove   List of triples to remove from resource.
+     * @param triplesToInsert   List of triples to add to resource.
+     * @param predicate         Predicate of relationship (assumed to be URI).    
+     * @param object            Object of relationship (assumed to be literal).    
+     * @return                  void
+     */
+    protected void updateDateTriple(QuadAcc triplesToRemove,
+                                    QuadDataAcc triplesToInsert,
+                                    String predicate,
+                                    String object) {
+        triplesToRemove.addTriple(new Triple(NodeFactory.createURI(""),
+                                             NodeFactory.createURI(predicate),
+                                             NodeFactory.createVariable("o" + String.valueOf(suffix))));
+        triplesToInsert.addTriple(new Triple(NodeFactory.createURI(""),
+                                             NodeFactory.createURI(predicate),
+                                             NodeFactory.createLiteral(object, XSDDatatype.XSDdateTime)));
+        suffix++;
+    }
+
+    /**
+     * Utility function for updating last modified date.  Current
+     * implementation utilizes a blank node.
+     *
+     * @param triplesToRemove   List of triples to remove from resource.
+     * @param triplesToInsert   List of triples to add to resource.
+     * @param object            Object of relationship (assumed to be literal).    
+     * @return                  void
+     */
+    protected void updateModifiedDate(QuadAcc triplesToRemove,
+                                      QuadDataAcc triplesToInsert,
+                                      String object) {
+        String eventPred = "http://www.loc.gov/premis/rdf/v1#hasEvent";
+        String eventTypePred = "http://www.loc.gov/premis/rdf/v1#hasEventType";
+        String eventDatePred = "http://www.loc.gov/premis/rdf/v1#hasEventDateTime";
+        Node bnode = NodeFactory.createAnon();
+
+        triplesToRemove.addTriple(new Triple(NodeFactory.createURI(""),
+                                             NodeFactory.createURI(eventPred),
+                                             NodeFactory.createVariable("o" + String.valueOf(suffix))));
+        triplesToInsert.addTriple(new Triple(NodeFactory.createURI(""),
+                                             NodeFactory.createURI(eventPred),
+                                             bnode));
+        triplesToInsert.addTriple(new Triple(bnode,
+                                             NodeFactory.createURI(eventTypePred),
+                                             NodeFactory.createLiteral("migration")));
+        triplesToInsert.addTriple(new Triple(bnode,
+                                             NodeFactory.createURI(eventDatePred),
+                                             NodeFactory.createLiteral(object, XSDDatatype.XSDdateTime)));
```

Comment:
It looks like the hasEventDateTime is the date of the "migration" event, not the date that the migrated version was updated.

http://id.loc.gov/ontologies/premis.html#hasEventDateTime

That said, it's not clear to me how, or whether to capture the last modification date for datastreams.